packages/api-client/src/store/security-schemes.ts

```diff
@@ -62,47 +62,47 @@ export function extendedSecurityDataFactory({
   }
 
   /** Delete a security scheme and remove the key from its corresponding parent */
-  const deleteSecurityScheme = (scheme: SecurityScheme) => {
+  const deleteSecurityScheme = (schemeUid: string) => {
     // Remove the scheme from any collections that reference it (should only be 1 collection)
     Object.values(collections).forEach((c) => {
-      if (c.securitySchemes.includes(scheme.uid)) {
+      if (c.securitySchemes.includes(schemeUid)) {
         collectionMutators.edit(
           c.uid,
           'securitySchemes',
-          c.securitySchemes.filter((s) => s !== scheme.uid),
+          c.securitySchemes.filter((s) => s !== schemeUid),
         )
       }
     })
 
     // Remove the scheme from any collections that use it
     Object.values(collections).forEach((c) => {
-      if (scheme.uid in c.auth) {
-        const { [scheme.uid]: toDelete, ...rest } = c.auth
+      if (schemeUid in c.auth) {
+        const { [schemeUid]: toDelete, ...rest } = c.auth
         collectionMutators.edit(c.uid, 'auth', rest)
       }
     })
```

Comment:
Can this not be done in 1 loop?

Reply:
yup, think it was written at two different times